tests/lit/single/pointer/ref_assignment_operator.st

```diff
@@ -0,0 +1,11 @@
+// RUN: (%COMPILE %s && %RUN) | %CHECK %s
+// CHECK: 5
+FUNCTION main : DINT
+    VAR
+        a : REF_TO DINT;
+        b : DINT := 5;
+    END_VAR
+    a REF= b;
+
+    printf('%d$N', a^);
+END_FUNCTION
```

Comment:
I'd like to have some correctness tests for more complex/error-prone types also (i.e. strings and/or structs)

Reply:
I've adjusted the `referenceto_variable_referencing_struct.st` file and added an additional test covering an array (very basic though), ping me if you want further tests